server/cluster/cluster.go

```diff
@@ -41,38 +40,30 @@ func NewServer(db db.ArgoDB, enf *rbac.Enforcer, cache *servercache.Cache, kubec
 	}
 }
 
-func (s *Server) getConnectionState(cluster appv1.Cluster, errorMessage string) (appv1.ConnectionState, string) {
+func (s *Server) getConnectionState(cluster appv1.Cluster, errorMessage string) appv1.ConnectionState {
 	if clusterInfo, err := s.cache.GetClusterInfo(cluster.Server); err == nil {
-		return clusterInfo.ConnectionState, clusterInfo.Version
+		return clusterInfo.ConnectionState
 	}
 	now := v1.Now()
 	clusterInfo := servercache.ClusterInfo{
-		ConnectionState: appv1.ConnectionState{
-			Status:     appv1.ConnectionStatusSuccessful,
-			ModifiedAt: &now,
-		},
+		ConnectionState: appv1.ConnectionState{},
 	}
 
-	config := cluster.RESTConfig()
-	config.Timeout = time.Second
-	version, err := s.kubectl.GetServerVersion(config)
-	if err != nil {
-		clusterInfo.Status = appv1.ConnectionStatusFailed
-		clusterInfo.Message = fmt.Sprintf("Unable to connect to cluster: %v", err)
-	} else {
-		clusterInfo.Version = version
+	if cluster.LastCacheSyncTime != "" {
```

Comment:
If the controller is down the cluster connection state will be obsolete. I think we should introduce `Unknown` status in addition to `Successful` and `Failed`. The `Unknown` status should be used is status attribute is missing or `modifiedAt` is too old.